firmware/quadruna/BMS/src/io/io_bootloader.c

```diff
@@ -0,0 +1,101 @@
+#include "io_bootloader.h"
+#include <stdint.h>
+#include "cmsis_gcc.h"
+#include "hw_hal.h"
+#include "hw_utils.h"
+
+#define BOOT_CAN_START 1012
+
+extern uint32_t __boot_code_start__;
+extern uint32_t __boot_code_size__;
+
+extern ADC_HandleTypeDef   hadc1;
+extern FDCAN_HandleTypeDef hfdcan1;
+extern TIM_HandleTypeDef   htim1;
+extern TIM_HandleTypeDef   htim3;
+extern TIM_HandleTypeDef   htim15;
+extern UART_HandleTypeDef  huart1;
+extern SPI_HandleTypeDef  hspi2;
+extern CRC_HandleTypeDef   hcrc;
+
+static void modifyStackPointerAndStartApp(const uint32_t *address)
```

Comment:
Honestly given that this piece of code here is being repeated in all 5 files. I would like this to be moved to shared

Reply:
The externs or the whole bootloader file?